### PR TITLE
Fix handling of Wii NAND's /tmp directory during savestate loads.

### PR DESCRIPTION
**Pull request by phire**
Fixes [issue 9496](https://bugs.dolphin-emu.org/issues/9496) and potentially other issues with savestates and wii games.

---
This change is [Reviewable](https://reviewable.io/reviews/dolphin-emu/dolphin/3798)

### CODE REVIEW

File: Source/Core/Core/IPC_HLE/WII_IPC_HLE.cpp [hunk 1]
```diff
@@ -270,6 +270,13 @@ void DoState(PointerWrap &p)
 	p.Do(reply_queue);
 	p.Do(last_reply_time);
 
+	if (p.GetMode() == PointerWrap::MODE_READ)
+	{
+		// We need to make sure all file handles are closed so WII_IPC_Devices_fs::DoState can successfully re-create /tmp
+		for (u32 i = 0; i<IPC_MAX_FDS; i++)
```
Comment (lioncash) [inline Source/Core/Core/IPC_HLE/WII_IPC_HLE.cpp]: Space before and after the `<`

**Comment by Parlane** [comment]
Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3798)*

**Comment by Parlane** [comment]
Review status: all files reviewed at latest revision, 1 unresolved discussion.

---

*[Source/Core/Core/IPC_HLE/WII_IPC_HLE.cpp, line 276 \[r1\]](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3798#-KGB4syD-CnkhBMcVWlx-r1-276:-KGdNWXUaZQYThQE_pEE:1814846539) ([raw file](https://github.com/dolphin-emu/dolphin/blob/147edd5bd0d3015954241347f821b2de80c6b97e/Source/Core/Core/IPC_HLE/WII_IPC_HLE.cpp#L276)):*
This has been fixed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/dolphin-emu/dolphin/3798)*

**Comment by Pringo** [comment]
This PR added four extra DLLs to the Windows download that weren't there before. Was that intentional?

**Comment by Parlane** [comment]
This PR added no DLLs. That'd be correlation not causation for ya.

**Comment by Pringo** [comment]
I mean 4.0-9257 compared to 4.0-9254.

**Pull request merged**